internal/pkg/bulk/engine.go

```diff
@@ -98,13 +110,136 @@ func NewBulker(es esapi.Transport, tracer *apm.Tracer, opts ...BulkOpt) *Bulker
 	}
 
 	return &Bulker{
-		opts:        bopts,
-		es:          es,
-		ch:          make(chan *bulkT, bopts.blockQueueSz),
-		blkPool:     sync.Pool{New: poolFunc},
-		apikeyLimit: semaphore.NewWeighted(int64(bopts.apikeyMaxParallel)),
-		tracer:      tracer,
+		opts:                  bopts,
+		es:                    es,
+		ch:                    make(chan *bulkT, bopts.blockQueueSz),
+		blkPool:               sync.Pool{New: poolFunc},
+		apikeyLimit:           semaphore.NewWeighted(int64(bopts.apikeyMaxParallel)),
+		tracer:                tracer,
+		remoteOutputConfigMap: make(map[string]map[string]interface{}),
+		// remote ES bulkers
+		bulkerMap:         make(map[string]Bulk),
+		remoteOutputLimit: semaphore.NewWeighted(1),
```

Comment:
Is this semaphore only to control updates? Or would it be better to replace this with a `RWLock`?

Reply:
It is only used when updating `bulkerMap`, as I've seen concurrency issues in the logs. 
I can replace it with a `sync.RWMutex`